client/my-sites/theme/controller.jsx

```diff
@@ -2,43 +2,25 @@
  * External Dependencies
  */
 import React from 'react';
-import omit from 'lodash/omit';
 import debugFactory from 'debug';
 import startsWith from 'lodash/startsWith';
-import i18n from 'i18n-calypso';
 
 /**
  * Internal Dependencies
  */
 import ThemeSheetComponent from './main';
 import ThemeDetailsComponent from 'components/data/theme-details';
 import { getCurrentUser } from 'state/current-user/selectors';
-import { getThemeDetails } from 'state/themes/theme-details/selectors';
 import {
 	receiveThemeDetails,
 	receiveThemeDetailsFailure,
 	setBackPath
 } from 'state/themes/actions';
 import wpcom from 'lib/wp';
 import config from 'config';
-import { decodeEntities } from 'lib/formatting';
 
 const debug = debugFactory( 'calypso:themes' );
-let themeDetailsCache = new Map();
-
-export function makeElement( ThemesComponent, Head, store, props ) {
```

Comment:
I still want to get rid of it for themes lists too, but that can wait for another PR.

It's just a layer of indirection we don't really need any more. Ideally, we want some sort of generic adapter that connects a component to a routing middleware, and I had originally envisaged `makeElement` to be that (when it contained even more logic). However, I don't think so anymore; e.g. its signature isn't really good -- for example, it shouldn't have to accept `Head` as an arg.

But yeah, moving it to where it is actually used is fine for now.